yaks/lib/yaks/default_policy.rb

```diff
@@ -17,13 +19,59 @@ def initialize(options = {})
       @options = DEFAULTS.merge(options)
     end
 
+    # Main point of entry for mapper derivation. Calls
+    # derive_mapper_from_collection or derive_mapper_from_item
+    # depending on the model.
+    #
     # @param model [Object]
     # @return [Class] A mapper, typically a subclass of Yaks::Mapper
     #
-    # @raise [NameError] only occurs when the model is anything but a collection.
+    # @raise [RuntimeError] occurs when no mapper is found or a collection
+    #                       is given but does not respond to :first
     def derive_mapper_from_object(model)
+      mapper = detect_configured_mapper_for(model)
+      return mapper if mapper
       return derive_mapper_from_collection(model) if model.respond_to? :to_ary
-      derive_mapper_from_single_object(model)
+      derive_mapper_from_item(model)
+    end
+
+    # Derives a mapper from the given collection.
+    #
+    # @param collection [Object]
+    # @return [Class] A mapper, typically a subclass of Yaks::Mapper
+    #
+    # @raise [RuntimeError] when collection does not respond to :first
+    def derive_mapper_from_collection(collection)
+      raise_not_a_collection(collection) unless collection.respond_to?(:first)
```

Comment:
Hmm, yeah, but do you think the user will ever call this method directly? For me, if the the collection responds to `#to_ary`, I wouldn't assume that it also responds to `#first`. If an object responds to `#to_ary`, it's probably using composition with an internal array, but maybe `#first` isn't implemented.